prototype/p2p/testing_utils.go

```diff
@@ -0,0 +1,43 @@
+package p2p
+
+import (
+	"crypto/rand"
+	"sync/atomic"
+)
+
+type fnCallStub struct {
+	called *int32 // act as bool
```

Comment:
Why the `act as` if it could be an int or a bool?

Reply:
Because of the atomic nature of it, as it is used in goroutines by multiple concurrent routines and I did not want to use `atomic.Bool` from uber's atomic lib.